pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/parser/InternalCalls.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2024 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.modeldefinition.parser;
+
+import hudson.Extension;
+import org.jenkinsci.plugins.pipeline.modeldefinition.model.CredentialsBindingHandler;
+import org.jenkinsci.plugins.pipeline.modeldefinition.withscript.WithScriptDescribable;
+import org.jenkinsci.plugins.workflow.cps.IgnoredInternalClasses;
+
+@Extension public final class InternalCalls implements IgnoredInternalClasses {
+
+    @Override public boolean ignore(Class<?> clazz) {
+        return WithScriptDescribable.class.isAssignableFrom(clazz) ||
+            CredentialsBindingHandler.class.isAssignableFrom(clazz) ||
+            clazz.getName().startsWith("org.jenkinsci.plugins.pipeline.modeldefinition.");
```

Comment:
Various classes in `src/main/groovy/`. Was not sure how to classify these more cleanly.